android/src/main/java/com/amplitude/android/Configuration.kt

```diff
@@ -39,8 +41,10 @@ open class Configuration @JvmOverloads constructor(
     var minTimeBetweenSessionsMillis: Long = MIN_TIME_BETWEEN_SESSIONS_MILLIS,
     var trackingSessionEvents: Boolean = true,
     override var identifyBatchIntervalMillis: Long = IDENTIFY_BATCH_INTERVAL_MILLIS,
-    override var identifyInterceptStorageProvider: StorageProvider = AndroidStorageProvider()
-) : Configuration(apiKey, flushQueueSize, flushIntervalMillis, instanceName, optOut, storageProvider, loggerProvider, minIdLength, partnerId, callback, flushMaxRetries, useBatch, serverZone, serverUrl, plan, ingestionMetadata, identifyBatchIntervalMillis, identifyInterceptStorageProvider) {
+    override var identifyInterceptStorageProvider: StorageProvider = AndroidStorageProvider(),
+    override var identityStorageProvider: IdentityStorageProvider = FileIdentityStorageProvider(),
+    val migrateLegacyData: Boolean = false,
+) : Configuration(apiKey, flushQueueSize, flushIntervalMillis, instanceName, optOut, storageProvider, loggerProvider, minIdLength, partnerId, callback, flushMaxRetries, useBatch, serverZone, serverUrl, plan, ingestionMetadata, identifyBatchIntervalMillis, identifyInterceptStorageProvider, identityStorageProvider) {
```

Comment:
Are we providing the ability to overwrite the identityStorageProvider here?

Reply:
yes, similar to `storageProvider` and `identifyInterceptStorageProvider`